lib/services/role.go

```diff
@@ -262,18 +262,40 @@ func (r *RoleV2) SetForwardAgent(forwardAgent bool) {
 
 // Check checks validity of all parameters and sets defaults
 func (r *RoleV2) CheckAndSetDefaults() error {
+	// make sure we have defaults for all fields
 	if r.Metadata.Name == "" {
 		return trace.BadParameter("missing parameter Name")
 	}
+	if r.Metadata.Namespace == "" {
```

Comment:
I changed `r.Spec.Namespaces`, `r.Spec.NodeLabels`, and `r.Spec.Resources` to only set defaults when `nil`.

I can't really do that here for Namespace without changing the underlying type for `r.Metadata.Namespace` since I can't distinguish not being set and being set to `""` without making it a pointer.

What situation would we want to allow an empty namespace to restrict access to nodes? In your above example you can remove the User if you don't want them to have access any nodes?